frontend/src/components/PlaidLinkButton.jsx

```diff
@@ -0,0 +1,48 @@
+import { useEffect, useState } from "react";
+import { usePlaidLink } from "react-plaid-link";
+import { getAuth } from "firebase/auth";
+
+const BACKEND_URL = import.meta.env.VITE_BACKEND_URL;
+
+const PlaidLinkButton = () => {
+  const [linkToken, setLinkToken] = useState(null);
+
+  useEffect(() => {
+    const fetchLinkToken = async () => {
+      const user = getAuth().currentUser;
+      if (!user) return;
+
+      const idToken = await user.getIdToken();
+
+      const res = await fetch(`${BACKEND_URL}/api/create_link_token`, {
+        method: "POST",
+        headers: {
+          "Content-Type": "application/json",
+          Authorization: `Bearer ${idToken}`,
+        },
+      });
+
+      const data = await res.json();
+      setLinkToken(data.link_token);
+    };
+
+    fetchLinkToken();
+  }, []);
```

Comment:
@fariasn516 does this need to be done in a useEffect?  could it be done on a click or nav event somewhere higher up?